### PR TITLE
:hammer: Adding DBCLocale option ?

### PR DESCRIPTION
**Pull request by Intemporel**

### CODE REVIEW

File: WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs [hunk 1]
```diff
@@ -166,7 +185,7 @@ private void Load(string filename, int id, int nameIndex, Dictionary<long, strin
             
             private void Load(string filename, int id, int nameIndex, Dictionary<long, long> dictionary)
             {
-                Load(filename, row => dictionary.Add(row.GetInt(id), row.GetInt(nameIndex)));
+                Load(filename, row => dictionary.Add(row.GetInt(id), row.GetInt(nameIndex + (int) dbcSettingsProvider.GetSettings().DBCLocale)));
```
Comment (funjoker) [inline WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs:188]: that makes 0 sense to me

File: WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs [hunk 2]
```diff
@@ -27,6 +27,25 @@ public enum DBCVersions
         SHADOWLANDS_41079 = 41079
     }
 
+    public enum DBCLocale
+    {
+        LANG_enUS = 0,
+        LANG_enGB = 1,
```
Comment (funjoker) [inline WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs:33]: enGB = enUS

File: WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs [hunk 3]
```diff
@@ -27,6 +27,25 @@ public enum DBCVersions
...
+        LANG_frFR = 3,
+        LANG_deDE = 4,
+        LANG_enCN = 5,
+        LANG_zhCN = 6,
+        LANG_enTW = 7,
+        LANG_zhTW = 8,
+        LANG_esES = 9,
+        LANG_esMX = 10,
+        LANG_ruRU = 11,
+        LANG_ptPT = 12,
```
Comment (funjoker) [inline WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs:44]: ptPT = ptBR

File: WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs (hunk 1 above)
Comment (BAndysc) [reply WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs:188]: it should be used in `Load(string filename, int id, int nameIndex, Dictionary<long, string> dictionary)`, a version which loads a string, not an int **AND** it is only required sometimes (not all strings have locale versions), so I'd recommend adding a bool parameter whether it should take a locale version or not (i.e. `Map` should use locale version, but `MapDirectory` should not)

Comment (funjoker) [reply WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs:188]: If you want german locale ,load german dbc.
If you want french locale, load french dbc.
etc.

Comment (BAndysc) [reply WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs:188]: > If you want german locale ,load german dbc. If you want french locale, load french dbc. etc.

not really, in the old dark times of Wrath (and cata? not sure here) there were 16 columns per string column for each locale, but only one is filled (others are empty)

Comment (Intemporel) [reply WoWDatabaseEditor.Common/WDE.DbcStore/DbcStore.cs]: i use french DBC, and the tool send me an error message when i want to open it. Now with that you can specify directly on DBC option the locale you want to use, and the Loader go into the right column for string localisation

**Comment by BAndysc** [comment]
Thanks for the contribution!

**Pull request merged**